### PR TITLE
fix: ready status of pod

### PR DESCRIPTION
**Pull request by kilosonc**

### CODE REVIEW

File: src/locales/en-US/pages.ts [hunk 1]
```diff
@@ -246,6 +246,10 @@ export default {
   'pages.cluster.podsTable.podStatus': 'Status',
   'pages.cluster.podsTable.containerName': 'Container name',
   'pages.cluster.podsTable.containerStatus': 'Container status',
+  'pages.cluster.podsTable.ready': 'Ready',
```
Comment (bysph) [inline src/locales/en-US/pages.ts:249]: Consider changing to “Ready or not” or “Ready status”

File: src/locales/zh-CN/pages.ts [hunk 2]
```diff
@@ -246,6 +246,10 @@ export default {
   'pages.cluster.podsTable.podStatus': '状态',
   'pages.cluster.podsTable.containerName': '容器名称',
   'pages.cluster.podsTable.containerStatus': '容器状态',
+  'pages.cluster.podsTable.ready': '就绪',
```
Comment (bysph) [inline src/locales/zh-CN/pages.ts:249]: Consider changing to “是否就绪” or “就绪状态”

File: src/pages/clusters/Pods/PodsTable/index.tsx [hunk 3]
```diff
@@ -623,7 +625,24 @@ export default (props: { data: CLUSTER.PodInTable[], cluster?: CLUSTER.Cluster |
...
       dataIndex: 'onlineStatus',
       key: 'onlineStatus',
-      render: (text: string) => status2StateNode.get(text),
+      render: (text: string) => (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+          {status2StateNode.get(text)}
+          {
+          text === 'unknown' && (
+          <Tooltip title={formatMessage('ready.unknown.description')}>
+            <QuestionCircleOutlined style={{ marginLeft: '6px' }} />
```
Comment (bysph) [inline src/pages/clusters/Pods/PodsTable/index.tsx:634]: Would it be better to set it as a warning color? Such as red or yellow.

File: src/pages/clusters/Pods/util.tsx [hunk 4]
```diff
@@ -338,16 +339,23 @@ export const refreshPodsInfo = (data?: CLUSTER.ResourceTree) => {
...
 
+        let readyCount = 0;
         let restartCount = 0;
         let onlineStatus = offlineState;
         if (containerStatuses && containerStatuses.length > 0) {
           restartCount = containerStatuses[0].restartCount;
           if (containerStatuses.length === containers.length) {
             onlineStatus = onlineState;
+            if (containers.map((c) => c.readinessProbe)
+              .filter((p) => p !== undefined && p !== null).length === containers.length) {
```
Comment (bysph) [inline src/pages/clusters/Pods/util.tsx]: Maybe you should use "<" here?

File: src/locales/en-US/pages.ts [hunk 5]
```diff
@@ -246,6 +246,10 @@ export default {
...
+  'pages.cluster.podsTable.ready.unknown.description': 'Readiness probe not configured yet, online status unknown',
+  'pages.cluster.podsTable.ready.ready': 'Ready',
```
Comment (ryanwuer) [inline src/locales/en-US/pages.ts]: ready.ready?

File: src/pages/clusters/Pods/util.tsx [hunk 6]
```diff
@@ -338,16 +339,23 @@ export const refreshPodsInfo = (data?: CLUSTER.ResourceTree) => {
...
+            if (containers.map((c) => c.readinessProbe)
+              .filter((p) => p !== undefined && p !== null).length < containers.length) {
+              onlineStatus = unknownState;
+            }
             containerStatuses.forEach(
               (containerStatus: any) => {
                 if (!containerStatus.ready) {
                   onlineStatus = offlineState;
+                } else {
+                  readyCount += 1;
```
Comment (iamyeka) [inline src/pages/clusters/Pods/util.tsx:358]: For me, i would use a flag called `ready` which has a default value: `true`, then judge if any container is not ready, set `ready` to false if any exist.

File: src/services/clusters.d.ts [hunk 7]
```diff
@@ -336,6 +336,7 @@ declare namespace CLUSTER {
     ip: string
     onlineStatus: string
     createTime: string
+    ready: number
```
Comment (iamyeka) [inline src/services/clusters.d.ts:339]: use a boolean type for better understanding?

**Review by ryanwuer — APPROVED** [review]

**Review by bysph — APPROVED** [review]

**Pull request merged**